### ISSUE TITLE
Auto-completing methods from not-in-scope blanket impls makes auto-complete hard to use

### ISSUE THREAD

**Issue by chklauser** [intellij-rust/intellij-rust#8901]
## Environment

* **IntelliJ Rust plugin version:** 0.4.171.4656-221
* **Rust toolchain version:** 1.61.0 (fe5b13d68 2022-05-18) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2022.1.2 (CL-221.5787.29)
* **Operating system:** Windows 11 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
When using the relatively popular crate `color-eyre`, auto-completion in IntelliJ Rust is significantly harder to use, because IntelliJ Rust includes methods from the [`OwoColorize` trait](https://docs.rs/owo-colors/3.4.0/owo_colors/trait.OwoColorize.html) (re-exported by `color-eyre`) on _every_ type.

Example:
![image](https://user-images.githubusercontent.com/149009/172004544-5b5fba04-d60c-4687-b784-8ca4e7d5536e.png)

This happens because the `owo_colors` crate defines a blanket implementation that covers all (sized) types

```rust
// TODO: figure out some wait to only implement for fmt::Display | fmt::Debug | ...
impl<D: Sized> OwoColorize for D {}
```

### Ideas 
 1. Use heuristics to ignore blanket impls/impls that are too broad? (might be difficult for the user to understand)
 2. Let the user define a denylist of module paths/type paths to ignore for the not-in-scope completion (might be difficult to discover)
 3. Change ranking of not-in-scope completions

## Steps to reproduce

 1. Set up a create with a dependency on `color-eyre` or `owo_colors`
 2. Hit `.` after any sized value --> the auto-completion window is filled with not-in-scope methods

I have created a tiny reproduction repo: https://github.com/chklauser/owo-autocomplete-repro

**Comment by vlad20012** [comment]
Hmm, blanket impls should already be ranked after at least inherent (non-trait) impls, and maybe after non-blanket trait impls (this is based on Machine Learning, so I can't say for sure how the ML model treats that fact that a trait impl is blanket, but we provide this fact to the ML model)

**Comment by chklauser** [comment]
Yes, the inherent methods are ranked at the top, thankfully 🙂, but I would wish that something like the [creation_flags](https://doc.rust-lang.org/std/process/struct.Command.html#method.creation_flags) method (from a `impl CommandExt for Command`) to be ranked higher.

**Comment by ivan-aksamentov** [comment]
This is still a problem and RustRover inherited it. Where and how should I report it?
